stock_partner_delivery_window/models/res_partner.py

```diff
@@ -152,3 +152,23 @@ def short_format_time(time):
                 )
             res[partner.id] = "\n".join(opening_times_description)
         return res
+
+    def copy_data(self, default=None):
+        result = super().copy_data(default=default)
+        not_time_windows = self.delivery_time_preference != "time_windows"
+        not_copy_windows = not_time_windows or "delivery_time_window_ids" in default
```

Comment:
We should probably manage the case where `not_time_windows` is `False and `default` is `None` ?

Not sure if it's makes sense could you add unit test to cover such case ?

Reply:
Thanks @petrus-v for your input. The test has been added to cover the case and the code is simplified. As expected result of `super().copy_data(default=default)` is of type `dict` (without any overriding: https://github.com/odoo/odoo/blob/14.0/odoo/models.py#L4540), the `result` is now used instead of `default`.

Reply:
Thanks :+1: